### DISCUSSION TITLE
required_filter: [field, ...]

### DISCUSSION THREAD

**Discussion by lloydtabb** [malloydata/malloy#342] [What's Next: Language Proposals]
In order to prevent really large queries, we need the ability to prevent queries from running without a filter on a specific field (or fields).

`required_filter: field`

or

`required_filter: [field, ...]`

LookML has `conditionally_filter`, which basically says that one of these fields must have a filter.  I'm not sure how often this feature is used.  I'm pretty sure it is not well understood.

**Comment by mtoy-googly-moogly** [comment]
Another approach would be some sort of a "sample filter" which picks a small subset of the data but disables itself if any real filters are placed on the source.
